### DISCUSSION TITLE
Some basic questions about Backrest/Restic

### DISCUSSION THREAD

**Discussion by gurabli** [garethgeorge/backrest#430] [Q&A]
Hi! 
First of all, thank you for the great work with Backrest. I just configured it on my home linux server to do backups to remote (and local) locations. Just testing SFTP (to Hetzner Storage Box) and it looks it is working really nice. 

However, since it is a backup that I would like to rely on, I do have some questions (even if some of the are n00b questions):

1. Question about the volumes.
```
      - ./backrest/cache:/cache
      - /MY-BACKUP-DATA:/userdata # [optional] mount local paths to backup here.
      - /MY-REPOS:/repos # [optional] mount repos if using local storage, not necessary for remotes e.g. B2, S3, etc.
```
**Cache**: this can be just a directory on my host, correct? I have it in the /opt/backrest at the moment.
**MY-BACKUP-DATA**: do I get this correctly, this is the SOURCE that Backrest container will see, from where I select the folders I would like to include in the backup. I have it now mapped to /mnt so I see all my files there. I did selected one folder for testing from here in the Plan -> Paths, like: /userdata/Pictures (which is actually /mnt/Pictures on my host. Is this correct?
**MY-REPOS**: I don't get the function of this. Is this the path to the local mount on my host (like an external USB drive mounted to host) if I want to do the backups TO this destination?

I'm sure I will have some more questions, but this is for start. And thanks to taking time to reading, answering these.

**Comment by garethgeorge** [comment]
Hey, glad you're liking the project so far! Your reading is right

 * Cache is any directory with sufficient storage for persistent caching of pack files from the repo (i.e. parts of the index, etc). Keeping this persistent across container restarts accelerates Backrest's ability to scan and deduplicate (without having to refetch the index of the last snapshot from your storage provider).
 *  /MY-BACKUP-DATA is just your documents folder or similarly anything you want to backup on your host machine.
 * /MY-REPOs - right , this is generally just included to remind users that if using local storage it _must_ be a location on the hostfs mounted in the container. I've had a few issue reports related to users sending all of their backups to a location that turns out to be inside the docker container's filesystem which is a docker footgun but an easy mistake to make. They then (very unfortunately) discover all of their backups are missing when the container is destroyed and recreated. If you're using SFTP you don't need to configure this mount.

Hope that clarifies a bit!

**Reply by gurabli** [reply]
Thanks for your detailed reply.

As for cache, I put it on my HDD not system SSD, I guess there is lot of write/read operations going on, so not good to have it on SSD.

My backups seam to work just fine. Tried restore too, went nice.
It looks that the docker container is using quite a lot of RAM, around 6-7GB. Is that actually normal?

SFTP is a bit slow, but that is more or less due to how SFTP works. Getting around 12-25 MB/s to Hetzner Germany. But, I get the same speeds with CIFS mounts too - I'll do a test what is Backrest speed if I use the CIFS mounted destination for backup.

**Reply by garethgeorge** [reply]
for what it's worth, I do actually use an SSD as my cache device -- but it's my understanding that restic is primarily using the cache for pack files that make up the index (e.g. it's understanding of the file tree + metadata). 

There's a separate `$TMPDIR` env var (that I should probably add to the compose file example) which is where restic writes partial blobs for upload -- that's more likely the write sensitive location to worry about.
